src/Ucommerce.Avenue.Umbraco/Controllers/PartialViewController.cs

```diff
@@ -15,24 +15,27 @@ public ActionResult CategoryNavigation()
         {
             var categoryNavigationModel = new CategoryNavigationViewModel();
 
-            ICollection<Category> rootCategories = CatalogLibrary.GetRootCategories().Where(x=>x.DisplayOnSite).ToList();
+            var catalogLibrary = ObjectFactory.Instance.Resolve<CatalogLibrary>();
+            IEnumerable<Category> rootCategories = catalogLibrary.GetRootCategories().Where(x=> x.Display).ToList();
 
             categoryNavigationModel.Categories = MapCategories(rootCategories);
 
             return View("/views/PartialView/CategoryNavigation.cshtml", categoryNavigationModel);
         }
 
-        private IList<CategoryViewModel> MapCategories(ICollection<Category> categoriesToMap)
+        private IList<CategoryViewModel> MapCategories(IEnumerable<Category> categoriesToMap)
         {
             var categoriesToReturn = new List<CategoryViewModel>();
 
             foreach (var category in categoriesToMap)
             {
                 var categoryViewModel = new CategoryViewModel();
 
-                categoryViewModel.Name = category.DisplayName();
-                categoryViewModel.Url = CatalogLibrary.GetNiceUrlForCategory(category);
-                categoryViewModel.Categories = MapCategories(CatalogLibrary.GetCategories(category).Where(x=>x.DisplayOnSite).ToList());
+                categoryViewModel.Name = category.DisplayName;
+                // categoryViewModel.Url = CatalogLibrary.GetNiceUrlForCategory(category);
+                
+                var catalogLibrary = ObjectFactory.Instance.Resolve<CatalogLibrary>();
```

Comment:
I don't really understand what is happening here. 
Couldn't the `Resolve` be called once outside the `foreach`?

Reply:
True, I haven't noticed, I did quick and dirty implementation. Good catch.

Reply:
ok, I thought I was getting something